### PR TITLE
Test Documentation and Scaffolding

### PR DESCRIPTION
**Pull request by brianb-sf**
## Overview
- I have created a base set of documentation we can discuss as a team along with the API call and backing data models we discussed. Please review and add comments.

## Pending Items
- [ ] Bonus features and requirements. We need to figure out what exactly we want these to be and have them added to the documentation and the designs cleared up.
- [ ] Team sign off on this provided scaffolding.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,71 @@
...
+
+Feel free to add more tests as you see fit but the above is the minimum requirement.
+
+## Design
+
+- See this [Figma File](https://www.figma.com/file/UOQDbU02GG2yaJMfrO9q9d/Flutter-Test?node-id=0%3A1) for design specifics like 
+fonts, themes, colors, etc.
+
+![List View](screenshots/listview.png)
+![Detail View](screenshots/detailview.png)
```
Comment (brianb-sf) [inline README.md:57]: chore: this needs to be updated with the new detail screen design once Caroline updates Figma.

File: app/lib/main.dart [hunk 2]
```diff
@@ -23,7 +24,25 @@ class HomePage extends StatelessWidget {
   Widget build(BuildContext context) {
     return Scaffold(
       body: Center(
-        child: Text('RestauranTour'),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            const Text('RestauranTour'),
```
Comment (jeffs-sf) [inline app/lib/main.dart:30]: chore: typo
```suggestion
            const Text('RestaurantTour'),
```

File: app/lib/respositories/yelp_repository.dart [hunk 3]
```diff
@@ -0,0 +1,107 @@
+import 'package:dio/dio.dart';
+import 'package:flutter/foundation.dart';
+import 'package:restaurantour/models/restaurant.dart';
+
+const _apiKey = '<PUT YOUR API KEY HERE>';
+
+class YelpRepository {
```
Comment (jeffs-sf) [inline app/lib/respositories/yelp_repository.dart:7]: question: do we want to provide tests for this Repository or do we want applicants to?

**Review by jeffs-sf — COMMENTED** [review]
Looking great, nice work @brianb-sf!

File: app/lib/main.dart (hunk 2 above)
Comment (brianb-sf) [reply app/lib/main.dart:30]: If you look at the designs the name of the app is actually `RestauranTour` 😄

File: app/lib/respositories/yelp_repository.dart (hunk 3 above)
Comment (brianb-sf) [reply app/lib/respositories/yelp_repository.dart:7]: I was thinking we could leave it up to the candidates to write tests. I don't want to establish a pattern for the tests because they might implement it differently and want to see what they come up with.

File: README.md [hunk 4]
```diff
@@ -1,10 +1,71 @@
-# Superformula Mobile Developer Coding Test
+# RestaurantTour
```
Comment (jeffs-sf) [inline README.md]: chore: this one should be updated to design then, haha.

File: app/lib/main.dart (hunk 2 above)
Comment (jeffs-sf) [reply app/lib/main.dart:30]: Oh... lol. Clever.

File: app/lib/respositories/yelp_repository.dart (hunk 3 above)
Comment (jeffs-sf) [reply app/lib/respositories/yelp_repository.dart:7]: I like that thinking.

File: README.md (hunk 4 above)
Comment (brianb-sf) [reply README.md]: Touché 😄

Comment (brianb-sf) [reply README.md]: Fixed

**Comment by ryan-sf** [comment]
**question**: thoughts on locking down library choices to a few top options or listing it as a bonus?  

For state management & DI:

1. [provider](https://pub.dev/packages/provider)
2. [Riverpod](https://pub.dev/packages/riverpod)
3. [bloc](https://pub.dev/packages/bloc)
4. [get_it](https://pub.dev/packages/get_it)/[get_it_mixins](https://pub.dev/packages/get_it_mixin)
5. [Mobx](https://pub.dev/packages/mobx)

My justification is twofold. 1) selfishly I'd prefer to not review submissions with one-off libs that we probably will not use. 2) I'd like to start building consistency in how we approach building new applications, and these are the most common libs that developers will see in tutorials.  As developers move between clients, having a consistent stack goes a long way.

You'll notice that I left off my personal favorite package, binder 😄

**Comment by jeffs-sf** [comment]
@ryan-sf I like the idea of locking it down for both of those points as well, but I don't think it'd be a bonus option. State management is pretty normal in Flutter so it should be expected to have to use something.

**Comment by brianb-sf** [comment]
@ryan-sf can you add the section to the README around state management?

**Comment by ryan-sf** [comment]
> @ryan-sf can you add the section to the README around state management?

Added a "Technical Requirements" section to capture this.  Also tweaked the "Code Values" section.

File: README.md [hunk 5]
```diff
@@ -1,22 +1,114 @@
...
+
+Please restrict your usage of state or dependency injection to the following options:
+
+1. [provider](https://pub.dev/packages/provider)
+2. [Riverpod](https://pub.dev/packages/riverpod)
+3. [bloc](https://pub.dev/packages/bloc)
+4. [get_it](https://pub.dev/packages/get_it)/[get_it_mixins](https://pub.dev/packages/get_it_mixin)
+5. [Mobx](https://pub.dev/packages/mobx)
+
+We ask this because this challenge values value consistency and efficiency over ingenuity.  Using commonly used libraries ensures that we can review your code in a timely manner and allows us to provide better feedback.
```
Comment (brianb-sf) [inline README.md]:
```suggestion
We ask this because this challenge values consistency and efficiency over ingenuity.  Using commonly used libraries ensures that we can review your code in a timely manner and allows us to provide better feedback.
```

**Review by ryan-sf — APPROVED** [review]
+10

**Review by russell-sf — APPROVED** [review]
+1 Looks good!
